examples/basic.rs

```diff
@@ -48,8 +45,9 @@ fn main() {
     println!("Shutting down session");
 
     RUNNING.store(false, Ordering::Relaxed);
-    session.shutdown();
-    let _ = reader.join();
+    session.shutdown()?;
+    let _ = reader.join().map_err(|err| wintun::Error::from(format!("{:?}", err)))?;
```

Comment:
nit: use `....map_err(|err| format!("{:?}", err).into())?`

Reply:
Can't use `....map_err(|err| format!("{:?}", err).into())?`, because the function's return value is `Result<(), Box<dyn std::error::Error>>`
![image](https://github.com/nulldotblack/wintun/assets/30760636/c3fb5831-a03d-4612-b912-eaf51b0284b5)

Reply:
Ah I see. Nevermind